### PR TITLE
fix(lmlayer): Polyfills for wordbreak support on oldest-supported platforms

### LINKED ISSUE #2323: [Android] Blank keyboard on Android 4.4.2
A user updated to Keyman 12.0.4206 stable but gets a blank SIL Euro Latin keyboard. All other Keyman keyboards also fail to load.

When connected with a test build, Chrome listed console errors
```
Uncaught SyntaxError: Unexpected identifier keyboard.html:270
Uncaught ReferenceError: Map is not defined keyman.js:18986
```

The SyntaxError is the use of `let` (ES6) and I think the browser for 4.4.2 is ES5.

The ReferenceError correponds to lm-layer's use of Map():
```
var PromiseStore = function() {
  this._promises = new Map();
```

Perhaps the webView on that version of Android doesn't support Map()?

I loaded Keyman for Android 11.0.2106 stable and that keyboard loaded fine.

### Configuration
Device: Samsung S4
OS: Android 4.4.2 (SDK 19)
Keyman: 12.0.4206
Keyboards: SIL Euro Latin (user also wanted SIL IPA and Greek)

### PR DESCRIPTION
**Pull request by jahorton**
Fixes #2323.

While fixing the IE unit-test targets at the same time isn't strictly necessary to get the Android devices fixed, it makes a great litmus test for things actually functioning on the oldest Android APIs we support - the `charCodeAt` issue reported was also a blocker for the now-fixed unit tests on IE.

While es6-shim could be used instead as a general polyfill for the worker, that seemed a bit overkill for use in the worker.

### CODE REVIEW

File: common/predictive-text/polyfills/array.from.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
+// From https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from
+// Any npm-based ones require use of `require`, which won't work for us.
+
+// Production steps of ECMA-262, Edition 6, 22.1.2.1
+if (!Array.from) {
```
Comment (jahorton) [inline common/predictive-text/polyfills/array.from.js]: Fixing the first issue wasn't enough for a full fix in IE; this polyfill was also necessary.  Note the following two links:
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from#Browser_compatibility
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/codePointAt#Browser_compatibility

`Array.from` requires a more modern Chrome for Android than `String.charCodeAt`... so it's almost certain this is also necessary for Android support.  (It also gets IE fully off the ground for the unit tests, so yay!)

**Comment by jahorton** [comment]
OK... so even more work's needed:
> that fixes codePointAt. :slightly_smiling_face:
> but now the next exception is
> blob:null/e6064721-5fca-410c-9563-5a6f01803574:Uncaught TypeError: Object opm has no method 'startsWith'

**Comment by darcywong00** [comment]
It's getting closer. Now the next exception involves [startsWith](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith)
```
blob:null/e6064721-5fca-410c-9563-5a6f01803574:Uncaught TypeError: Object opm has no method 'startsWith'
```

I hope it's not going to be a long list of functions to fill for the version 30.0 of Chrome for Android on those old devices.

**Comment by jahorton** [comment]
> It's getting closer. Now the next exception involves [startsWith](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith)
> 
> ```
> blob:null/e6064721-5fca-410c-9563-5a6f01803574:Uncaught TypeError: Object opm has no method 'startsWith'
> ```
> 
> I hope it's not going to be a long list of functions to fill for the version 30.0 of Chrome for Android on those old devices.

If it starts getting too long, I'll swap the fix to just using the es6-shim polyfill.  It's a 58 kB blob when minified; definitely a substantial block of code, but it's not exactly the worst thing we could use.

**Comment by darcywong00** [comment]
I took the liberty of adding the polyfill, and am pleased to report on the older Android device (API 19):
1. Keyboard loads
2. nrc.en.mtnt 0.1.3 works and inserts suggestions

**Comment by mcdurdin** [comment]
Tested on iPhone locally. Looks good -- no errors loading, and all keyboard/lmlayer functions appear to be working correctly.

**Comment by jahorton** [comment]
Sweet.  I'll get it 🍒-picked to master tomorrow unless someone beats me to it.

**Pull request merged**